File: src/main/java/com/egodoy/alten/challenge/model/Room.java
package com.egodoy.alten.challenge.model;

import lombok.*;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(schema = "public", name = "room")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@Builder
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String roomView;

    @OneToOne
    private RoomType roomType;

    @Column
    private Short status;

    @Where(clause = "status = 1")
    @OneToMany(mappedBy = "room")
    private List<ReservationRoom> reservationRooms;

}

File: alten-ui-app/src/app/model/room.ts
export interface Room{
    id: number;
    roomView: string;
    roomType: {
        id: number;
        name?: string;
        occupants?: number;
        beds?: [
            {
                quantity: number;
                bedType: string
            },
            {
                quantity: number;
                bedType: string
            }
        ]
    }
    status?: number;
}